src/commands/branch.rs

```diff
@@ -61,7 +61,14 @@ pub struct BranchDeleteArgs {
 /// preceded by a "+". For information about branches, see
 /// https://github.com/martinvonz/jj/blob/main/docs/branches.md.
 #[derive(clap::Args, Clone, Debug)]
-pub struct BranchListArgs;
+pub struct BranchListArgs {
+    /// Show branches which local/remote targets are in the given revisions.
+    ///
+    /// If any of local/remote targets matched, the branch (including
+    /// non-matching targets) will be listed.
+    #[arg(long, short)]
+    revisions: Option<RevisionArg>,
```

Comment:
For better or worth, the `-r` argument to Git push takes `Vec<RevisionArgs>`. I think this should be consistent. (In fact, maybe `log` should do the same? One reason for it is that `|` needs to be escaped in shells).

Reply:
Yeah, `log` can take multiple `-r` flags. This is just copied from `log` as I consider `branch list` is a log-like command for branches.

Reply:
I think you may have said can when you meant cannot. Log does not take multiple -r, and I think both commands should. It doesn't necessarily have to be resolved in this pr, though.

Reply:
Oops, I meant `log` could take multiple `-r`s, but it's not implemented right now.